### DISCUSSION TITLE
This is pretty unreadable, demonstrates rationale against using TypeScript

### DISCUSSION THREAD

**Discussion by mikegwhit** [vuejs/pinia#1322] [Polls]
Was encountering an error during development trying to follow docs failed so I dug through code and nothing going on here is readable. 

The only thing Pinia seems to logically accomplish is to act as an object with watchers on it as far as I can tell.  The docs don't reflect clear thinking has gone into this, so why is the author profiled wearing a head-mic like this guy spends more time at speaking engagements than being a practicing engineer???

Don't author huge public facing libraries that break backwards compatibility and be held like an innocent party.  You're costing people money.  This is garbage code and so far seems like a fraudulent library.

https://github.com/vuejs/pinia/blob/5b0c92cc592fbea910cc5b78f50d87dfb0162b10/packages/pinia/src/store.ts#L818-L845

**Comment by pikax** [comment]
I can read the code. Sometimes backwards compatibility need to be broken to make improvements.

**Comment by yyx990803** [comment]
Hey, nobody asked you to use it and you didn't pay for it either. As a user, the very least you can do is to communicate in a respectful and constructive manner, yet you failed to do that. The complexity of types in the source code is there to provide better intellisense for day-to-day usage, which is an obvious trade-off. You are projecting your frustration with your own incompetence in TypeScript on to the author of the library that is trying to help you for free. You are the embodiment of open source consumer entitlement and you are welcome to **fuck off** and never use this library again.

P.S. you have been banned from interacting with any repositories under the vuejs organization, forever. Please don't bother replying. Actually, I believe it'd be much better off if you just stop using open source and write everything yourself from scratch!
